process_lang8.py

```diff
@@ -0,0 +1,76 @@
+import random
+import argparse
+from os.path import join as pjoin
+
+'''
+convert lang8 corpus into 2 text files containing parallel verb phrases
+(original and corrected).
+
+about half the examples don't have corrections; can either ignore or add those as identity mappings
+
+currently doesn't do additional preprocessing besides that done by tajiri et al.
+'''
+
+def parse_entries(lines):
+    par_count = 0
+    original = list()
+    corrected = list()
+    for line in lines:
+        cols = line.split('\t')
+        if len(cols) < 6:
+            continue
+        else:
+            # NOTE prints examples with multiple corrections
+            #if len(cols) > 6:
+                #print(' ||| '.join(cols[4:]))
+            # NOTE not using multiple corrections to avoid train and dev having same source sentences
+            for corr_sent in cols[5:6]:
+                if cols[4] == corr_sent:
+                    continue
+                original.append(cols[4])
+                corrected.append(corr_sent)
+                par_count += 1
+    print('%d parallel examples' % par_count)
+    return original, corrected
+
+def write_text(split, data, part, lang8_path):
+    with open(pjoin(lang8_path, 'entries.%s.%s' % (split, part)), 'w') as fout:
+        fout.write('\n'.join(data))
+
+def process_data(lang8_path, split, dev_split_fract):
+    with open(pjoin(lang8_path, 'entries.%s' % split)) as fin:
+        lines = fin.read().strip().split('\n')
+        print('%d lines total in %s split' % (len(lines), args.split))
```

Comment:
The `args` variable is not defined in the `process_data method`, only in the main.